vendor/orchestra-event-observer/src/observer/mod.rs

```diff
@@ -0,0 +1,730 @@
+    if !config.is_authorization_required() {
+        let mut hook_formation = HookFormation::new();
+        if let Some(ref mut initial_hook_formation) = config.initial_hook_formation {
+            hook_formation
+                .stacks_chainhooks
+                .append(&mut initial_hook_formation.stacks_chainhooks);
+            hook_formation
+                .bitcoin_chainhooks
+                .append(&mut initial_hook_formation.bitcoin_chainhooks);
+        }
+        config.operators.insert(None, hook_formation);
+    }
+
+    loop {
+        let command = match observer_commands_rx.recv() {
+            Ok(cmd) => cmd,
+            Err(e) => {
+                if let Some(ref tx) = observer_events_tx {
+                    let _ = tx.send(ObserverEvent::Error(format!("Chanel error: {:?}", e)));
+                }
+                continue;
+            }
+        };
+        match command {
+            ObserverCommand::Terminate => {
+                if let Some(ref tx) = observer_events_tx {
+                    let _ = tx.send(ObserverEvent::Info("Terminating event observer".into()));
+                    let _ = tx.send(ObserverEvent::Terminate);
+                }
+                break;
+            }
+            ObserverCommand::PropagateBitcoinChainEvent(chain_event) => {
+                for event_handler in event_handlers.iter() {
+                    event_handler.propagate_bitcoin_event(&chain_event).await;
+                }
+                // process hooks
+                if config.hooks_enabled {
+                    let bitcoin_chainhooks = config
+                        .operators
+                        .values()
+                        .map(|v| &v.bitcoin_chainhooks)
+                        .flatten()
+                        .collect();
+
+                    let hooks_to_trigger = evaluate_bitcoin_chainhooks_on_chain_event(
+                        &chain_event,
+                        bitcoin_chainhooks,
+                    );
+                    let mut proofs = HashMap::new();
+                    for (_, transaction, block_identifier) in hooks_to_trigger.iter() {
+                        if !proofs.contains_key(&transaction.transaction_identifier.hash) {
+                            let rpc = Client::new(
+                                &format!(
+                                    "{}:{}",
+                                    config.bitcoin_node_rpc_host, config.bitcoin_node_rpc_port
+                                ),
+                                Auth::UserPass(
+                                    config.bitcoin_node_username.to_string(),
+                                    config.bitcoin_node_password.to_string(),
+                                ),
+                            )
+                            .unwrap();
+                            let txid = Txid::from_str(&transaction.transaction_identifier.hash)
+                                .expect("Unable to retrieve txid");
```

Comment:
Since this function already returns a `Result, would it be easy to convert all of these `expect`s into just returning an error instead?

Reply:
Addressed in https://github.com/hirosystems/clarinet/pull/402/commits/ce821313ed66b9190826dd147cdcc477cf6908f0

Reply:
Addressed in https://github.com/hirosystems/clarinet/pull/402/commits/ce821313ed66b9190826dd147cdcc477cf6908f0

Reply:
There are other expects in this function that could have the same treatment.

Reply:
Unless I'm missing something, the only expect I can see in this function are for optional resolving to none.
Note that `expect` and `try` will have the same effect, this function is basically a runloop, so interrupting the execution results in a process exit.